### PR TITLE
Add precommit hook so it works with precommit

### PR DESCRIPTION
**Pull request by milind-shakya-sp**
- [x] Make it work precommit, so it can be run in the commit stage or overriden to run in manual stage, which helps in CI checks in builds. Fixes https://github.com/brettcannon/caniusepython3/issues/197

### CODE REVIEW

**Comment by brettcannon** [comment]
How are people to specify what the hook is supposed to check? I.e. there's no requirements file, metadata file, or list of packages to test, so the CLI will error out. Should this have a line like:
```yaml
files: '*requirements*.txt'
```
or something?

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (brettcannon) [inline .pre-commit-hooks.yaml]: What's this for?

**Comment by milind-shakya-sp** [comment]
@brettcannon it would need [precommit](https://pre-commit.com/) to work. We can provide some details like the following.

```
You need to have precommit setup to use this hook.
Install Pre-commit and the commit-msg hook-type.

pip install pre-commit==1.11.1
pre-commit install
```

File: README.md [hunk 2]
```diff
@@ -80,6 +80,36 @@ various sources of help. If you want a specific starting point there are
...
+pre-commit install
+```
+
+It is best used along with [pre-commit](https://pre-commit.com/) You can use it along with pre-commit by adding the following hook in your ``.pre-commit-config.yaml`` file.
+
+    - repo: https://github.com/brettcannon/caniusepython3
+      sha: 'b97a4cd7'
+      hooks:
+          - id: caniusepython3
+            files: requirements.txt  # Update to match your requirements files accordingly.
```
Comment (milind-shakya-sp) [inline README.md]: @brettcannon yes, they would have to override this accordingly to how their requirements files are named.

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (milind-shakya-sp) [reply .pre-commit-hooks.yaml]: I removed it.

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- id: caniusepython3
+  name: caniusepython3
+  entry: caniusepython3
+  language: python
+  stages: [commit]
+  files: requirements.*\.txt$
```
Comment (milind-shakya-sp) [inline .pre-commit-hooks.yaml]: @brettcannon Runs on requirements files.

Comment (brettcannon) [inline .pre-commit-hooks.yaml]:
```suggestion
  files: requirements\.txt$
```

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  description: This script takes in a set of dependencies and then figures out which of them are holding you up from porting to Python3.
```
Comment (brettcannon) [inline .pre-commit-hooks.yaml]:
```suggestion
  description: 'caniusepython3: figure out what packages are holding you up from porting to Python 3.'
```

**Comment by milind-shakya-sp** [comment]
@brettcannon thanks for the suggestions. They make sense to me.

**Comment by milind-shakya-sp** [comment]
Looks like the errors are due to https://github.com/brettcannon/caniusepython3/issues/173

**Comment by vsajip** [comment]
Sorry - certificate error now fixed. I can't yet set up Let's Encrypt certificates to auto-renew on this site :-(

**Comment by milind-shakya-sp** [comment]
Thank you @vsajip

**Pull request merged**